### ISSUE TITLE
Array/object confusion

### ISSUE THREAD

**Issue by jmeas** [chbrown/rfc6902#2]
This might be a known bug but the diff fails to differentiate objects and arrays. This can be shown in the [browser test](http://chbrown.github.io/rfc6902/).

```js
// input 
{
  "repositories": [
    "amulet",
    "flickr-with-uploads"
  ]
}

// output
{
  "repositories": {}
}
```

You would expect this to be a replace operation, but instead it's two removals. This leaves you with an empty *array*, and not an empty object.

**Comment by chbrown** [comment]
You're totally right. My Array vs. Object checking is broken because:

    var a = [1, 2, 3, 5, 8, 13, 21];
    Object(a) === a; // true!?

A fix is forthcoming.

**Comment by chbrown** [comment]
Fixed in distributable as of `62d9d20` and published to npm as `v1.0.3`.

**Comment by jmeas** [comment]
dope
